### ISSUE TITLE
FGSM loss computation

### ISSUE THREAD

**Issue by gongzhitaao** [openai/cleverhans#104]
According to the paper [Adversarial examples in the physical world](https://arxiv.org/abs/1607.02533), FGSM uses `y_true`, the true label, to generate the adversarial images, instead of model predictions.  Besides, it does not really make sense to me to use `y_pred` and logits to compute the loss since `y_pred = softmax(logits)`.  Instead of 

```python
def fgsm(x, predictions, eps, clip_min=None, clip_max=None)
```

The second parameter should be
```python
def fgsm(x, y_true, eps, clip_min=None, clip_max=None)
```

**Comment by goodfeli** [comment]
We should document this clearly in the docstring, but we now recommend against using FGSM with `y_true` due to the label leaking effect discovered in "Adversarial Machine Learning at Scale."

**Comment by gongzhitaao** [comment]
Interesting.  We just observed label leaking too on various datasets. Especially on STL10, with adversarial training, the accuracy for a simple model could easily get over 90%. LOL
